### DISCUSSION TITLE
Git Hooks: auto-run a script, e.g. prior to commit

### DISCUSSION THREAD

**Discussion by FObersteiner** [imk-toolkit/imk-toolkit#78] [Ideas]
If you ever failed CI tests just because you forgot to run the formatter: here's a way to avoid this. Use [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks). I just found out about them ;-)

For the above example you can use e.g. the [pre-commit framework](https://pre-commit.com/) to make sure `black` formatting etc. is run automatically before you commit something.

**Comment by ucyo** [comment]
Yes. Git hooks are actually pretty nice. How about a running process ... I made some additions to the Makefile here #79 check them out.. Maybe those are interesting ..
